HERACLES-3 and HERACLES-4: Updated Event Clinical Cardiac and Phone Call Follow-up forms

marta- opened the pull request.

marta- committed:
HERACLES-3: Implement Event Clinical Cardiac form feedback

marta- committed:
HERACLES-4: Implement Phone Call Follow-up form feedback

sashaandjic reviewed (approved):
tested

The special section edit was done in the Hospital visit only for "Another procedure not on the list":
![image](https://github.com/data-team-uhn/cards/assets/23243244/4958ba15-02ec-4dd5-9559-64ca6a510304)

marta- commented:
@Hajany this branch includes the updated Physical Assessments form which was already  merged.

Hajany commented:
Conditions for _Ejection Fraction %_ and _Coronary Artery Calcium Score (CAC)_ are not working as expected
![image](https://github.com/data-team-uhn/cards/assets/96542282/f6ccb2e2-799b-44f1-99c1-587a3cf15a21)

acrowthe commented:
Ejection fraction appears to be working as I'd expect based on the specifications - one instance of the question shows up based on the specifications in the attached image, a second instance shows up based on `heart failure` as specified further down in the specifications.

Coronary Artery Calcium Scores is not conditional which does not match the specifications

sdumitriu commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Event - Clinical Cardiac.xml:
```diff
@@ -339,16 +237,16 @@
 				<property>
 					<name>label</name>
-					<value>PR222222</value>
+					<value>Post TA</value>
```
Why not Post Triggered Assesment?

sdumitriu committed:
[misc] Remove extra newlines in question text

acrowthe commented:
Noticed an inconsistency: Some instances of the value for `angiography (diagnostic only)` contain a single space before the bracket `(` while others contain two spaces:
```
<value>Angiography  (diagnostic only)</value>
<value>Angiography (diagnostic only)</value>
```
Looks like `ec_24` and all questions that are conditional on `ec_24` use a double space while `ec_28` and related questions use a single space. As a result it shouldn't cause any immediate issues but could lead to issues further along in data processing

acrowthe committed:
HERACLES-3: Implement Event Clinical Cardiac form feedback
Fix missing Coronary Artery Calcium Score conditional

acrowthe commented:
Coronary Artery Calcium Score has had it's current condition added to the form however there's a flaw in how the condition is implemented. Per specifications:
```
"if selected Coronary Artery Disease (CAD) OR Non_CV OR None of the Above in 1"
AND
"if selected cardiac_ct and/or angiography (diagnostic only) for 1.4.4"
```
This was translated into:
```
(ec_10="CAD" OR ec_10=NonCV OR ec_10="None of the above")
AND
(ec_24 includes "Cardiac CT" OR ec_24  includes "Angiography  (diagnostic only)")
```
However, `ec_24` is conditional on `ec_10="CAD"` so `ec_10=NonCV OR ec_10="None of the above"` will never have a matching `ec_24` result.

I believe the correct condition for Coronary Artery Calcium Score (`ec_43`) should be:
```
(ec_10="CAD" OR ec_10=NonCV OR ec_10="None of the above")
AND
(
    ec_24 includes "Cardiac CT" OR ec_24 includes "Angiography  (diagnostic only)"
    OR
    ec_28_noncv includes "Cardiac CT" OR ec_28_noncv includes "Angiography (diagnostic only)"
)
```

Ejection fraction (`ec_42`) should receive a similar treatment:
```
(ec_10="CAD" or ec_10=NonCV or ec_10="None of the above")
AND
(
    ec_24  includes "Stress echo"
    OR
    ec_28_noncv includes "Stress echo"
)
```

acrowthe committed:
HERACLES-3: Implement Event Clinical Cardiac form feedback
- Fix CAC and Ejection Fracction conditions for other cases
- Fix extra spaces in Angiagraphy (diagnostic only) values
